pkg/kubelet/kubelet_node_status.go

```diff
@@ -360,8 +357,8 @@ func (kl *Kubelet) syncNodeStatus() {
 	if kl.kubeClient == nil || kl.heartbeatClient == nil {
 		return
 	}
-	if kl.registerNode {
-		// This will exit immediately if it doesn't need to do anything.
+	// Register to apiserver only on the first start when registerNode is true.
+	if kl.registerNode && !kl.registrationCompleted{
```

Comment:
What's the benefit of this refactoring? Maybe the old code is a better one as it check and update `registrationCompleted` only inside `registerWithAPIServer`. If you really wanted to extract `registrationCompleted` from `registerWithAPIServer` to `syncNodeStatus`, you can let `registerWithAPIServer` returns a bool value that indicated whether registration is successful or not, and update  `registrationCompleted` inside `syncNodeStatus` too. By this you can check and update `registrationCompleted` only inside `syncNodeStatus`.